pkg/services/ngalert/ngalert.go

```diff
@@ -172,3 +175,30 @@ func (ng *AlertNG) IsDisabled() bool {
 	}
 	return !ng.Cfg.IsNgAlertEnabled()
 }
+
+// getRuleDefaultIntervalSeconds returns the default rule interval if the interval is not set.
+// If this constant (1 minute) is lower than the configured minimum evaluation interval then
+// this configuration is returned.
+func (ng *AlertNG) getRuleDefaultIntervalSeconds() int64 {
+	ruleMinIntervalSeconds := ng.getRuleMinIntervalSeconds()
+	if defaultIntervalSeconds < ruleMinIntervalSeconds {
+		return ruleMinIntervalSeconds
+	}
+	return defaultIntervalSeconds
+}
+
+// getRuleMinIntervalSeconds returns the configured minimum rule interval.
+// If this value is less or equal to zero or not divided exactly by the scheduler interval
+// the scheduler interval (10 seconds) is returned.
+func (ng *AlertNG) getRuleMinIntervalSeconds() int64 {
+	if ng.Cfg.UnifiedAlertingMinInterval <= 0 {
+		return defaultBaseIntervalSeconds // if it's not configured; apply default
```

Comment:
This is already very hard to follow as is - do you think we can simplify this?

I'm thinking that both `ng.Cfg.UnifiedAlertingMinInterval <= 0` and `ng.Cfg.UnifiedAlertingMinInterval%defaultBaseIntervalSeconds` should be a hard error now at a configuration level, that way we can always make the assumption that the minimum internal is always going be >= 10 and will be divisible by the base default.

Technically, erroring out will be a breaking change at a configuration level but a very obvious one as opposed to causing surprises at runtime which sends us on a wild goose chase on issues while we try to reproduce. 

cc: @torkelo

Reply:
https://github.com/grafana/grafana/discussions/39709